### ISSUE TITLE
Search Highlighter zeichnet auch src-Attribute von Bildern aus

### ISSUE THREAD

**Issue by alexwenz** [FriendsOfREDAXO/search_it#186]
Die Suche nach Winter und Übergabe des Suchbegriffes via ?search_highlighter=winter erzeugt folgenden Quellcode:

`<img src="/images/overlay/<span class='searched-it'>winter</span>1.jpg" alt="" width="60" height="16">`

**Comment by tyrant88** [comment]
Es liegt an dem HTML-tag im Attribut, da matched der Reguläre Audruck wieder, obwohl er nicht sollte.
gestern 4 RegEx Ansätze probiert. 
Muss ich wohl mit einer ganz anderen Logik lösen..

**Comment by alexplusde** [comment]
Kannst du es in 2 Etappen lösen? Erst ersetzen, dann dort filtern, wo sie zu viel sind? Als Lösungsansatz

**Comment by tyrant88** [comment]
Ja so ähnlich ist der Gedanke.

**Comment by tyrant88** [comment]
Nächster Versuch. Alex meint es geht jetzt Alex. Willst du auch noch mal testen?

**Comment by alexplusde** [comment]
Wenn Alex das sagt, dann vertraue ich darauf. Alex' haben generell einen guten Ruf ;)

**Comment by tyrant88** [comment]
Ja, wenn Kevin das gesagt hätte würde ich nicht soviel drauf geben... :-)

**Comment by danspringer** [comment]
Darf ich mich hier noch einklinken? 
Der Highlighter zeichnet auch Treffer z.B. in einer <textarea> aus, die bereits mit Werten aus einem Artikel vorbelegt ist, was zu HTML in den Inputs eines Formulars führt.

**Comment by alexplusde** [comment]
`<button></button>` dürfte auch so ein Kandidat dafür sein.

**Comment by tyrant88** [comment]
Hab leider gerade null Zeit

**Comment by xong** [comment]
Als Anregung: Ich kann man mir vorstellen, dass man ähnlich wie im Plaintext-Plugin mit dem _[PHP Simple HTML DOM Parser](https://sourceforge.net/projects/simplehtmldom/)_ arbeitet. Dort hatte ich für RexSearch eine `remove`-Methode implementiert. Wahrscheinlich kann man ähnlich auch eine "highlight-replace"-Methode schreiben, die nur Inhalte bestimmter Tags parst und gefundene Suchbegriffe hervorhebt.

Alternativ könnte man auch das Highlighting mit Javascript umsetzen und z. B. _[mark.js](https://github.com/julmot/mark.js/)_ verwenden. Das erscheint mir die beste und performanteste Lösung zu sein.

**Comment by alexwenz** [comment]
Javascript wäre sicher eine gute Lösung, hätte aber den Nachteil, dass man das im Frontend einbinden und die schöne "seach_it" macht alles autark Situation aufgeben müsste ...

**Comment by xong** [comment]
Der Ansatz war eigentlich immer, dass die Ausgabe ins Modul gehört und das Addon vielfältige Funktionen und Optionen anbietet, um dabei zu unterstützen. Das "Search Highlighter"-Plugin wurde übrigens auch nie von mir programmiert, weil es nicht ins Konzept gepasst hat. Es kam quasi von außen.
